src/main/java/user_interface/login_leaderboard/RegisterPanel.java

```diff
@@ -13,8 +15,17 @@ public class RegisterPanel extends Panel implements ActionListener {
     private final JTextField username;
     private final JPasswordField password;
     private final JTextField email;
+    private String passedUsername;
+    private String passedEmail;
+    private String passedPassword;
+
+    RegisterUserController registerUserController;
+
+    public RegisterPanel(RegisterUserController registerUserController) {
+
+        this.registerUserController = registerUserController;
+
 
-    public RegisterPanel() {
         this.build();
         JLabel askUserName = new JLabel("Please enter your username");
         // check if user already exists, if not continue
```

Comment:
on the UI I can see the "please enter your username" but the askEmail line is grey and I can't see the askPassword line (because there is no bounds line?)
Is there a problem with labelFormat?

Reply:
Fixed, it was because I forgot to pass in the askEmail JLabel into the labelSet array :D